libkpod/image/copy.go

```diff
@@ -152,9 +158,20 @@ func PullImage(store storage.Store, imgName string, allTags, quiet bool, sc *typ
 				}
 			}
 		}
-	} else if splitArr[0] == "oci" {
-		// needs to be implemented in future
-		return errors.Errorf("oci not supported")
+	} else if splitArr[0] == "oci-archive" {
+		// generate image reference with specified image:tag
+		if len(options.OciImage) > 0 {
+			srcRef, err = ociarchive.NewReference(splitArr[len(splitArr)-1], options.OciImage)
```

Comment:
This is only weakly related I guess, not new code in this PR, and not used by the new code in this PR:

I can’t see how the `for _, image := range images` loop at the end of the function works: it is iterating through `image` and setting the _destination_ reference, but the loop keeps `srcRef` unchanged. Or am I missing something?

Reply:
The for loop is for a docker-archive file that has more than image. It goes through the manifest.json and retrieves the names of the images saved and copies it over. I used the same srcRef because the images were coming from the same file.
I believe it doesn't successfully copy the multiple images stored in a docker-archive file, but rather it copies the same image and saves it as the various image names. I think we decided to fix this later to support multiple images.